### PR TITLE
Navigation: Ensure command palette is correctly translated

### PR DESCRIPTION
**Pull request by joshhunt**
**What is this feature?**

Rather than keep remembering to translate the nav tree in each place we render it, instead we should translate it once off when we load it into the store, that way anyone who uses it will just automatically get it translated.

**Why do we need this feature?**

So Grafana is more consistently translated.

**Who is this feature for?**

Everyone 🚀 

**Special notes for your reviewer**:

There's somehow still some places where we get an un-translated nav model (see secondary nav):

image

I'm also not enthusiastic about how the nav tree is stored _twice_ in the store, and having to translate it twice? Perhaps we could store just the nested navTree and use selectors to recursively find a node for a specific ID to replace the navIndex store?

### CODE REVIEW

File: public/app/app.ts [hunk 1]
```diff
@@ -121,6 +121,10 @@ export class GrafanaApp {
       setPanelDataErrorView(PanelDataErrorView);
       setLocationSrv(locationService);
       setTimeZoneResolver(() => config.bootData.user.timezone);
+
+      // We must wait for translations to load because some preloaded store state requires translating
+      await initI18nPromise;
```
Comment (joshhunt) [inline public/app/app.ts:126]: Alternatively, maybe we could await this further down like we used to, and dispatch an action later on (but before any react rendering) to translate/initialize the nav?

**Review by ashharrison90 — COMMENTED** [review]
think this makes sense. the untranslated content in the section nav is due to us not fully assigning the new translated content to the tree in `reducers/navModel.ts`. the ol' object reference problem 😄 

if you make `buildNavIndex` something like this and reassign the `parentItem`s children:
```typescript
function buildNavIndex(navIndex: NavIndex, children: NavModelItem[], parentItem?: NavModelItem) {
  const translatedChildren: NavModelItem[] = [];
  for (const node of children) {
    const translatedNode: NavModelItem = {
      ...node,
      text: getNavTitle(node.id) ?? node.text,
      subTitle: getNavSubTitle(node.id) ?? node.subTitle,
      emptyMessage: getNavTitle(node.emptyMessageId),
      parentItem: parentItem,
    };

    if (translatedNode.id) {
      navIndex[translatedNode.id] = translatedNode;
    }

    if (translatedNode.children) {
      buildNavIndex(navIndex, translatedNode.children, translatedNode);
    }
    translatedChildren.push(translatedNode);
  }
  if (parentItem) {
    parentItem.children = translatedChildren;
  }

  navIndex['not-found'] = { ...buildWarningNav('Page not found', '404 Error').node };
  navIndex['error'] = { ...buildWarningNav('Page error', 'An unexpected error').node };
}
```

everything should be translated correctly. not saying we _should_ do this, but hopefully it'll help you untangle it a bit 👍

**Review by joshhunt — COMMENTED** [review]
✅  I can't approve my own PR, again!

image

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

**Pull request merged**